### PR TITLE
Fix deprecation in Numeric

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Numeric.php [hunk 1]
```diff
@@ -561,7 +561,7 @@ public function isDiffChangeAllowed($object, $params = [])
      */
     public function isEmpty($data)
     {
-        return strlen($data) < 1;
+        return empty($data) || strlen($data) < 1;
```
Comment (jdreesen) [inline models/DataObject/ClassDefinition/Data/Numeric.php]: Not sure if this is right, as `"0"` is considered empty (see: https://3v4l.org/mgSdd).

Comment (dpfaffenbauer) [reply models/DataObject/ClassDefinition/Data/Numeric.php]: true, lets wait on how we resolve that on the other PR

Comment (dpfaffenbauer) [reply models/DataObject/ClassDefinition/Data/Numeric.php]: I think we should check if $data is actually a number, if not it is invalid.

Comment (brusch) [inline models/DataObject/ClassDefinition/Data/Numeric.php]:
```suggestion
        return is_numeric($data);
```
This should do the trick in this case.

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/DataObject/ClassDefinition/Data/Numeric.php [hunk 2]
```diff
@@ -561,7 +561,7 @@ public function isDiffChangeAllowed($object, $params = [])
      */
     public function isEmpty($data)
     {
-        return strlen($data) < 1;
+        return is_numeric($data);
```
Comment (brusch) [inline models/DataObject/ClassDefinition/Data/Numeric.php]:
```suggestion
        return !is_numeric($data);
```

**Comment by dpfaffenbauer** [comment]
thanks @brusch

**Pull request merged**